src/metabase/query_processor/middleware/constraints.clj

```diff
@@ -5,33 +5,46 @@
    [metabase.models.setting :as setting]
    [metabase.util.i18n :refer [deferred-tru]]))
 
-(def ^:private ^:const default-max-results-bare-rows 2000)
+(def ^:private ^:const default-max-unaggregated-query-row-limit 2000)
+(def ^:private ^:const default-max-aggregated-query-row-limit 10000)
 
-;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-results-bare-rows]])
+;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-unaggregated-query-row-limit]])
 ;; to a setting in 0.43 the setting, which allows for DB local value, can still be nil, so any places below that used
 ;; to reference the former constant value have to expect it could return nil instead
-(setting/defsetting max-results-bare-rows
+(setting/defsetting max-unaggregated-query-row-limit
```

Comment:
Unfortunately it's not this simple to change a setting's name. If the setting is already set on any instances, changing the name will essentially "unset" it since the `setting` table uses the setting's name as the key. In addition, since this has `:database-local :allowed`, it might have additional DB-specific values in the `metabase_database.settings` JSON field. (Looks like this could be the case for [SQLServer](https://github.com/metabase/metabase/pull/19267), not sure about others)

If we absolutely need a new env var, we either need
* A migration to change the setting name anywhere it might already exist in the app DB (and a rollback, preferably)
* Or, a new setting with a custom getter which migrates the value of `max-results-bare-rows` when it's fetched

Reply:
It also might be a product question if we want to continue allowing this setting to be set at the DB level, and what should happen if it's _also_ set via env var. The way database-local settings currently work is that they'll return the DB-local value preferentially to the env var value.

Reply:
@noahmoss backwards compatibility was my thought too when I picked this up. When I talked to @luizarakaki about it, it sounded to me like given the fact that the current setting doesn't really work, we are confident that it's not being used and that we can remove it without fear: https://metaboat.slack.com/archives/C064EB1UE5P/p1699472623388119

Reply:
Oh okay. In that case I'll stamp. 

I'm still a little concerned about the SQLServer case since that's the one database where it seems like there's an explicit mechanism to change the row limit when you're setting up the DB